server/src/oauth/db.rs

```diff
@@ -0,0 +1,423 @@
+
+/// Deletes a token for a DID
+pub async fn delete_token(pool: &PgPool, did: &str) -> cja::Result<()> {
+    sqlx::query!(
+        r#"
+        DELETE FROM oauth_tokens
+        WHERE did = $1
+        "#,
+        did
+    )
+    .execute(pool)
+    .await?;
+
+    Ok(())
+}
+
+/// Updates the display name for a token
+pub async fn update_token_display_name(
+    pool: &PgPool,
+    did: &str,
+    display_name: &str,
+) -> cja::Result<()> {
+    sqlx::query!(
+        r#"
+        UPDATE oauth_tokens
+        SET display_name = $2, updated_at_utc = NOW()
+        WHERE did = $1
+        "#,
+        did,
+        display_name
+    )
+    .execute(pool)
+    .await?;
+
+    Ok(())
+}
+
+/// Updates the handle for a token
+pub async fn update_token_handle(pool: &PgPool, did: &str, handle: &str) -> cja::Result<()> {
+    sqlx::query!(
+        r#"
+        UPDATE oauth_tokens
+        SET handle = $2, updated_at_utc = NOW()
+        WHERE did = $1
+        "#,
+        did,
+        handle
+    )
+    .execute(pool)
+    .await?;
+
+    Ok(())
+}
+
+/// Gets the most recent DPoP nonce for a DID
+pub async fn get_latest_nonce(pool: &PgPool, did: &str) -> cja::Result<Option<String>> {
+    // Find the most recent session for this DID that has a nonce
+    let row = sqlx::query!(
+        r#"
+        SELECT data FROM oauth_sessions 
+        WHERE did = $1 
+        ORDER BY updated_at_utc DESC 
+        LIMIT 1
+        "#,
+        did
+    )
+    .fetch_optional(pool)
+    .await?;
+
+    if let Some(row) = row {
+        let data: serde_json::Value = row.data.unwrap_or(serde_json::Value::Null);
+        let nonce = data
+            .get("dpop_nonce")
+            .and_then(|v| v.as_str())
+            .map(String::from);
+
+        return Ok(nonce);
+    }
+
+    Ok(None)
+}
+
+/// Updates a session's DPoP nonce
+pub async fn update_session_nonce(pool: &PgPool, session_id: Uuid, nonce: &str) -> cja::Result<()> {
+    // First get the current session data
+    let row = sqlx::query!(
+        r#"
+        SELECT data FROM oauth_sessions WHERE session_id = $1
+        "#,
+        session_id
+    )
+    .fetch_optional(pool)
+    .await?;
+
+    if let Some(row) = row {
+        let mut data: serde_json::Value = row.data.unwrap_or(serde_json::Value::Null);
+
+        // Update the nonce in the data
+        if let Some(obj) = data.as_object_mut() {
+            obj.insert(
+                "dpop_nonce".to_string(),
+                serde_json::Value::String(nonce.to_string()),
+            );
+        }
+
+        // Update the session with the new data
+        sqlx::query!(
+            r#"
+            UPDATE oauth_sessions
+            SET data = $1, updated_at_utc = NOW()
+            WHERE session_id = $2
+            "#,
+            data,
+            session_id
+        )
+        .execute(pool)
+        .await?;
+    }
+
+    Ok(())
+}
+
+#[allow(dead_code)]
+pub async fn cleanup_expired_sessions(pool: &PgPool) -> cja::Result<u64> {
```

Comment:
Don't we have a job that does this? Does it not use this method? Cleanup this method if we don't need it, or call it from the job that uses it